File: Top 100 Codes/02-Working with Numbers/14-Maximum number of handshakes/Maximumnumberofhandshakes.cpp
#include <iostream>
using namespace std; 

int main() {
    // Declare a variable to store the number of people
    int num;

    // Take user input for the number of people
    cout << "Enter the number of people: ";
    cin >> num;

    // Ensure the number of people is non-negative
    if (num < 0) {
        cout << "Number of people cannot be negative." << endl;
        return 1; // Exit with an error code
    }

    // Calculate the total number of handshakes using the formula nC2 = n * (n - 1) / 2
    int total = num * (num - 1) / 2;

    // Display the result
    cout << "For " << num << " people, there will be " << total << " handshakes." << endl;

    return 0;
}

File: Top 100 Codes/02-Working with Numbers/14-Maximum number of handshakes/Maximumnumberofhandshakes.py
# Program to calculate the maximum number of handshakes

# Take user input for the number of people
N = int(input("Enter the number of people: "))

# Ensure the input is non-negative
if N < 0:
    print("Number of people cannot be negative.")
else:
    # Calculate the number of handshakes using the formula: nC2 = n * (n - 1) / 2
    no_of_handshakes = int(N * (N - 1) / 2)

    # Print the result
    print(f"Maximum number of handshakes possible for {N} people is {no_of_handshakes}.")
